modules/virt-temporary-token-VNC.adoc

```diff
@@ -0,0 +1,64 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virt-accessing-vm-consoles.adoc
+
+:_content-type: PROCEDURE
+[id="virt-temporary-token-VNC_{context}"]
+= Generating a temporary token for VNC console
+
+Generate a temporary token to access the VNC of a virtual machine (VM).
+
+.Prerequisites
+
+* A running Windows virtual machine
+* ?
+
+.Procedure
+
+. Enable the feature gate in the `HCO` custom resource:
++
+[source,yaml]
+----
+$ oc patch hyperconverged kubevirt-hyperconverged -n {CNVNamespace} \
+ --type json -p '[{"op": "replace", "path": \
+ "/spec/featureGates/deployVmConsoleProxy", \
+ "value": true}]'
+# ...
+----
+
+. Generate a token:
++
+[source,terminal]
+----
+$ curl --header "Authorization: Bearer ${TOKEN}" \
+     "https://api.<cluster_fqdn>/apis/token.kubevirt.io/v1alpha1/namespaces/<namespace>/virtualmachines/<vm_name>/vnc?duration=<duration>" <1>
+----
+<1> Duration can be in hours and/or minutes. Example: `5h30m`
++
+Sample output:
++
+[source,terminal]
+----
+{ "token": "eyJhb..." }
+----
+
+. Use the token provided in the output to create a variable:
++
+[source,terminal]
+----
+$ export VNC_TOKEN="<token>"
+----
+
+.  Log in to the cluster:
++
+[source,terminal]
+----
+$ oc login --token ${VNC_TOKEN}
+----
+
```

Comment:
I've used the `oc login` in my demo showing this feature, because it was the simplest way to test that it works.

I expect that the token access will be used primarily by developers, that will use the token it their application. They may use `virtctl vnc` just for testing that it works.

Reply:
I mean that `oc login` changes the global login credentials that are used by subsequent `oc` and `virtctl` commands. User may not want to change the global credentials to a token that has only access to the VNC of one VM.
I've done it in the demo, because it was simple, but I think it is not the main use case for this feature.

> * Update the step "`Log in to the cluster`" to something similar to "`Log in to the cluster to access the VNC console of the VM`" using `oc login`.

I leave this change up to you, I don't have any opinion about which text is better.

> * Update the "Use virtctl to access the VNC console..." step to "Use virtctl to test the VNC console..." and place it in under Verification instead of Procedure.

This change sounds good, because using `virtctl vnc` with the token is not the main use case.